PWGHF/HFL/HFmAnalysis.cxx

```diff
@@ -0,0 +1,335 @@
+#include "Common/DataModel/EventSelection.h"
+#include <THashList.h>
+#include <vector>
+#include <memory>
+#include <THnSparse.h>
+#include "TDatabasePDG.h"
+
+using namespace o2;
+using namespace o2::aod;
+using namespace o2::framework;
+
+namespace o2::aod
+{
+namespace dqanalysisflags
+{
+DECLARE_SOA_COLUMN(IsEventSelected, isEventSelected, int);
+}
+
+DECLARE_SOA_TABLE(EventCuts, "AOD", "EVENTCUTS", dqanalysisflags::IsEventSelected);
+} // namespace o2::aod
+
+using MyCollisions = o2::soa::Join<aod::Collisions, aod::EvSels>;
+using MyMcCollisions = o2::soa::Join<aod::Collisions, aod::McCollisionLabels, aod::EvSels>;
+using MyEventsSelected = o2::soa::Join<aod::Collisions, aod::EvSels, o2::aod::EventCuts>;
+using MyMcEventsSelected = o2::soa::Join<aod::Collisions, aod::McCollisionLabels, aod::EvSels, o2::aod::EventCuts>;
+using MyMuons = o2::soa::Join<aod::FullFwdTracks, aod::McFwdTrackLabels>;
+
+constexpr static uint32_t gkMuonFillMap(VarManager::ObjTypes::Muon | VarManager::ObjTypes::MuonCov);
+constexpr static uint32_t gkEventFillMap(VarManager::ObjTypes::BC | VarManager::ObjTypes::Collision);
+constexpr static uint32_t gkTrackMCFillMap(VarManager::ObjTypes::ParticleMC);
+
+void DefineHistograms(HistogramManager*, TString);
+
+struct EventSelection {
+  Configurable<bool> fConfigTriggerUsed{"cfgtriggerUsed", false, "whether apply the software trigger"};
+  Configurable<bool> fConfigVtxZUsed{"cfgVtxZUsed", false, "whether apply the VtxZ trigger"};
+  Configurable<float> fConfigSoftwareTrigger{"cfgsoftwaretrigger", VarManager::kIsMuonSingleLowPt7, "softerware tigger flag"};
+
+  Configurable<float> fConfigVtxZMIN{"cfgVtxZMIN", -10., "min. z of primary vertex [cm]"};
+  Configurable<float> fConfigVtxZMAX{"cfgVtxZMAX", 10., "max. z of primary vertex [cm]"};
+
+  HistogramManager* fHistMan;
+  OutputObj<THashList> fOutputList{"output"};
+
+  float* fValues;
+  AnalysisCompositeCut* fEventCut;
+
+  void init(o2::framework::InitContext&)
+  {
+    VarManager::SetDefaultVarNames();
+    fValues = new float[VarManager::kNVars];
+    fEventCut = new AnalysisCompositeCut("event selection", "Event Selection", true);
+
+    AnalysisCut fCut;
+    if (fConfigVtxZUsed)
+      fCut.AddCut(VarManager::kVtxZ, fConfigVtxZMIN, fConfigVtxZMAX, false);
+    if (fConfigTriggerUsed)
+      fCut.AddCut(fConfigSoftwareTrigger, 0.5, 1.5, false);
+    fEventCut->AddCut(&fCut);
+
+    fHistMan = new HistogramManager("analysisHistos", "aa", VarManager::kNVars);
+    fHistMan->SetUseDefaultVariableNames(kTRUE);
+    fHistMan->SetDefaultVarNames(VarManager::fgVariableNames, VarManager::fgVariableUnits);
+
+    DefineHistograms(fHistMan, "EventBeforeCuts;EventAfterCuts;");
+    VarManager::SetUseVars(fHistMan->GetUsedVars());
+    fOutputList.setObject(fHistMan->GetMainHistogramList());
+  }
+
+  Produces<aod::EventCuts> eventSel;
+  template <uint32_t TEventFillMap, typename TEvent>
+  void runEventSel(TEvent const& event, aod::BCs const& bcs)
```

Comment:
Separate `Produces` from the function with a blank line.
Add documentation of the function.